osu.Game/Screens/SelectV2/BeatmapSetPanel.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Diagnostics;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Pooling;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Game.Beatmaps;
+using osu.Game.Graphics.Sprites;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.SelectV2
+{
+    public partial class BeatmapSetPanel : PoolableDrawable, ICarouselPanel
+    {
+        public const float HEIGHT = CarouselItem.DEFAULT_HEIGHT * 2;
+
+        [Resolved]
+        private BeatmapCarousel carousel { get; set; } = null!;
+
+        private Box activationFlash = null!;
+        private OsuSpriteText text = null!;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            Size = new Vector2(500, HEIGHT);
+            Masking = true;
+
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    Colour = Color4.Yellow.Darken(5),
+                    Alpha = 0.8f,
+                    RelativeSizeAxes = Axes.Both,
+                },
+                activationFlash = new Box
+                {
+                    Colour = Color4.White,
+                    Blending = BlendingParameters.Additive,
+                    Alpha = 0,
+                    RelativeSizeAxes = Axes.Both,
+                },
+                text = new OsuSpriteText
+                {
+                    Padding = new MarginPadding(5),
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreLeft,
+                }
+            };
+
+            Selected.BindValueChanged(value =>
+            {
+                activationFlash.FadeTo(value.NewValue ? 0.2f : 0, 500, Easing.OutQuint);
+            });
+
+            KeyboardSelected.BindValueChanged(value =>
+            {
+                if (value.NewValue)
+                {
+                    BorderThickness = 5;
+                    BorderColour = Color4.Pink;
+                }
+                else
+                {
+                    BorderThickness = 0;
+                }
+            });
+        }
+
+        protected override void PrepareForUse()
+        {
+            base.PrepareForUse();
+
+            Debug.Assert(Item != null);
+            Debug.Assert(Item.IsGroupSelectionTarget);
+
+            var beatmapSetInfo = (BeatmapSetInfo)Item.Model;
+
+            text.Text = $"{beatmapSetInfo.Metadata}";
+
+            this.FadeInFromZero(500, Easing.OutQuint);
+        }
+
+        protected override bool OnClick(ClickEvent e)
+        {
+            carousel.CurrentSelection = Item!.Model;
```

Comment:
To continue the theme above, this weirds me out. Locally the set panel select itself but then that gets hijacked to beatmap selection at `Carousel` level? It's a bit strange, is it supposed to be like that?

Reply:
Not too sure. How would you see it working? I'm trying to keep the knowlege of "what selects what" all in one place (`BeatmapCarousel`) and keep these panels as simple as possible. The other direction would be to make them very self aware.

For this specific case I guess it can already do that by setting its first available beatmap, which sounds fine and maybe preferable, until we add something like a recommendation step to the process. Once we have an external entity choosing which beatmap should be selected, I think it's going to be better to have this selection logic in `BeatmapCarousel`.

Reply:
I dunno... maybe at least an inline comment or something that this gets magically handled elsewhere?

The other suggestion would be to invert the flow and just give panels a delegate to request selection from the carousel.

Reply:
Hold that though until I get groups working, because selection is a bit obtuse there and i'm still finding a solution.

Reply:
I'll get back to this later then... if I remember to I guess :sweat_smile: